azurerm/internal/services/containers/resource_arm_kubernetes_cluster.go

```diff
@@ -120,6 +120,42 @@ func resourceArmKubernetesCluster() *schema.Resource {
 				},
 			},
 
+			"identity_profile": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kubelet_identity": {
```

Comment:
there's not really any value in nesting these fields - so it might be worth ditching the wrapper `identity_profile` to make this:

```
azurerm_kubernetes_cluster.test.kubelet_identity.0.user_assigned_identity_id
```

WDYT?

Reply:
Yes, I don't see the value for now either. The reason why I left it there was to reflect the api structure and to be prepared for extra elements in `identity_profile`.

I'm not sure why the AKS guys nested it, it might have to do with the possibility to assign multiple identities in the future, who knows 😄

Reply:
I pushed it anyways and will check if the acceptance test is passing